### PR TITLE
Added code to Display Limited Users for "Be a part of Drifty Family!

### LINKED ISSUE #252: Decluttering Home Page: Avoid Displaying All Users
### Is your feature request related to a problem? Please describe.

Showing all the users on landing page makes the home page cluttered.

image

### Describe the solution you'd like.

I present you 2 options

1. Create a separate page for contributors and add a link for the page in the footer.

image

2. Show only **top 5** contributors with **more** button which when clicked will direct to the this [link](https://github.com/SaptarshiSarkar12/Drifty/graphs/contributors) 

Don't worry about this 

image

I will find a way to add this.

### Screenshots

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by PriyansuMaurya**
## Fixes issue
Fixes #252 

## Changes proposed

1. Added code to Display Limited no of Users for "Be a part of Drifty Family!" section.
2. Added Little bit of Responsive improvements in "Be a part of Drifty Family!" section.

## Check List (Check all the applicable boxes) 
- [x] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] This PR does not contain plagiarized content.
- [x] The title of my pull request is a short description of the requested changes.

## Screenshots
image

scaling-telegram-4prwp6p6xgjfq79q-3000 app github dev_ (1)

scaling-telegram-4prwp6p6xgjfq79q-3000 app github dev_

### CODE REVIEW

**Review by SaptarshiSarkar12 — CHANGES_REQUESTED** [review]
@PriyansuMaurya I am requesting some changes on the following :
- Please add this animation for **43+** circular icon.

    [Screencast from 2023-08-24 20-29-24.webm](https://github.com/SaptarshiSarkar12/Drifty/assets/105960032/ea5e2968-b067-4d48-81f9-7984e885e2e1)

- Revert the animation changes you have made for the **contribution**, **stars** and **forks** buttons for _**mobile screens**_. Keep the same animation (violet as background colour when the buttons are hovered) for both desktop and mobile screens.
- Increase the size of the **contribution**, **stars** and **forks** buttons for mobile screens.
    drifty-e6krvubkg-saptarshisarkar12 vercel app_(Samsung Galaxy S20 Ultra)
- Change the style of the the **contribution**, **stars** and **forks** buttons to `rounded-lg` instead of `rounded-full`.
- For **IPad Mini**, the avatars are about to overlap. Please fix that.   
    drifty-e6krvubkg-saptarshisarkar12 vercel app_(iPad Mini)

File: Website/app/Contribute.js [hunk 1]
```diff
@@ -27,16 +28,24 @@ export default function Contribute( {props} ) {
...
             </div>
             <div className="md:w-1/2">
-            <h2 className="text-5xl sm:text-4xl font-bold md:mt-20 md:mb-20 md:pl-20 xs:text-center xs:mb-10" >Be a part of Drifty Family!</h2>
-                <div className="grid lg:grid-cols-7 md:grid-cols-5 sm:grid-cols-8 xs:grid-cols-4 space-x-0 gap-y-4 justify-items-center xs:pb-10">
-                    {props.contrib.map((item,index) => {
-                        return <a  href={item.html_url} key={index}><Image width={64} height={64} src={item.avatar_url + ".webp&s=64"} alt={item.login}></Image></a>
+            <h2 className="text-5xl sm:text-4xl my-12 font-bold lg:mt-20 md:mt-10 sm:pt-10 sm:mb-10 md:pl-20 xs:p-5 xs:text-center" >Be a part of Drifty Family!</h2>
+                <div className="grid lg:grid-cols-7 md:grid-cols-5 sm:grid-cols-8 xs:grid-cols-4 rounded-sm	 space-x-0 gap-y-4 justify-items-center xs:pb-10">
+                    {props.contrib.map((item, index) => {
+                        return index < 6 && <a href={item.html_url} key={index}><Image className="rounded-full" width={80} height={80} src={item.avatar_url + ".webp&s=64"} alt={item.login}></Image></a>
```
Comment (SaptarshiSarkar12) [inline Website/app/Contribute.js:34]: @PriyansuMaurya The dimensions of the fetched GitHub avatar needs to match with the dimension of the container. So, just commit the below change.
```suggestion
                        return index < 6 && <a href={item.html_url} key={index}><Image className="rounded-full" width={80} height={80} src={item.avatar_url + ".webp&s=80"} alt={item.login}></Image></a>
```

File: Website/app/Contribute.js [hunk 2]
```diff
@@ -27,16 +28,24 @@ export default function Contribute( {props} ) {
...
-                    <a className="md:rounded-full xs:p-2 lg:p-2 lg:w-52  border-2 md:border-white text-center   hover:text-black xs:transition ease-in-out delay-150 bg-blue-500 hover:-translate-y-1 hover:scale-110 hover:bg-indigo-500 duration-300" href="https://github.com/SaptarshiSarkar12/Drifty/stargazers"><i aria-hidden className="fa fa-regular fa-star pr-1"></i>Stars : {values[0]}</a>
-                    <a className="md:rounded-full xs:p-2 lg:p-2 lg:w-52 border-2 md:border-white text-center   hover:text-black xs:transition ease-in-out delay-150 bg-blue-500 hover:-translate-y-1 hover:scale-110 hover:bg-indigo-500 duration-300" href="https://github.com/SaptarshiSarkar12/Drifty/network/members"><i aria-hidden className="fa fa-duotone fa-code-fork pr-1"></i>Forks : {values[1]}</a>
+
+                <div className="flex flex-cols-3 justify-center xs:gap-7   lg:pl-5 text-lg font-semibold pr-4">
```
Comment (SaptarshiSarkar12) [inline Website/app/Contribute.js:45]: @PriyansuMaurya Change the flex to grid with 3 columns.
```suggestion
                <div className="grid grid-cols-3 justify-center gap-4 pl-3 pr-3 text-lg font-bold w-auto h-auto">
```

**Review by SaptarshiSarkar12 — CHANGES_REQUESTED** [review]
Please make this changes and let me know when you are done.

**Comment by PriyansuMaurya** [comment]
@SaptarshiSarkar12, I have made the requested changes.

**Review by SaptarshiSarkar12 — CHANGES_REQUESTED** [review]
@PriyansuMaurya Please make the buttons to `rounded-lg`.
Also, fix the text overflow in mobile screens : 
image

**Comment by SaptarshiSarkar12** [comment]
@PriyansuMaurya Also, revert the avatars from square to rounded shape as present in your first commit. The circular shape looks good.

**Review by SaptarshiSarkar12 — CHANGES_REQUESTED** [review]
@PriyansuMaurya Make the view something like this :
image

**Review by SaptarshiSarkar12 — CHANGES_REQUESTED** [review]
@PriyansuMaurya Please make the `rounded-lg` type of buttons for all displays and not just for mobile screens.
image

**Review by SaptarshiSarkar12 — CHANGES_REQUESTED** [review]
@PriyansuMaurya Please change all the `class` attributes to `className` attribute.

File: Website/app/Contribute.js [hunk 3]
```diff
@@ -27,21 +27,21 @@ export default function Contribute({ props }) {
...
                     {props.contrib.map((item, index) => {
-                        return index < 6 && <a  href={item.html_url} key={index}><Image width={64} height={64} src={item.avatar_url + ".webp&s=64"} alt={item.login}></Image></a>
+                        return index < 6 && <a href={item.html_url} key={index}><Image className="rounded-full" width={64} height={64} src={item.avatar_url + ".webp&s=64"} alt={item.login}></Image></a>
                     })}
                     <a href="https://github.com/SaptarshiSarkar12/Drifty/graphs/contributors" target="_blank" >
-                        <div className="h-16 w-16 flex items-center justify-center border-2 md:border-white text-center   hover:text-black xs:transition ease-in-out delay-150 bg-blue-500 hover:-translate-y-1 hover:scale-110 hover:bg-indigo-500 duration-300">
-                            +{totalNoOfContributors - 6}
+                        <div className="rounded-full h-16 w-16 flex items-center justify-center border-2 md:border-white text-center   hover:text-black xs:transition ease-in-out delay-150 bg-blue-500 hover:-translate-y-1 hover:scale-110 hover:bg-indigo-500 duration-300">
```
Comment (SaptarshiSarkar12) [inline Website/app/Contribute.js:36]: @PriyansuMaurya Please commit this below change :
```suggestion
                        <div className="rounded-full h-20 w-20 text-xl font-semibold flex items-center justify-center border-2 md:border-white text-center   hover:text-black xs:transition ease-in-out delay-150 bg-blue-500 hover:-translate-y-1 hover:scale-110 hover:bg-indigo-500 duration-300">
```

File: Website/app/Contribute.js [hunk 4]
```diff
@@ -27,21 +27,21 @@ export default function Contribute({ props }) {
                 </p>
             </div>
             <div className="md:w-1/2">
-            <h2 className="text-5xl sm:text-4xl my-12 font-bold lg:mt-20 md:mt-10 sm:pt-10 sm:mb-10 md:pl-20 xs:p-5 xs:text-center" >Be a part of Drifty Family!</h2>
+                <h2 className="text-5xl sm:text-4xl my-12 font-bold lg:mt-20 md:mt-10 sm:pt-10 sm:mb-10 md:pl-20 xs:p-5 xs:text-center" >Be a part of Drifty Family!</h2>
                 <div className="grid lg:grid-cols-7 md:grid-cols-5 sm:grid-cols-8 xs:grid-cols-4 rounded-sm	 space-x-0 gap-y-4 justify-items-center xs:pb-10">
                     {props.contrib.map((item, index) => {
-                        return index < 6 && <a  href={item.html_url} key={index}><Image width={64} height={64} src={item.avatar_url + ".webp&s=64"} alt={item.login}></Image></a>
+                        return index < 6 && <a href={item.html_url} key={index}><Image className="rounded-full" width={64} height={64} src={item.avatar_url + ".webp&s=64"} alt={item.login}></Image></a>
```
Comment (SaptarshiSarkar12) [inline Website/app/Contribute.js:33]: @PriyansuMaurya Please also commit this changes :
```suggestion
                        return index < 6 && <a href={item.html_url} key={index}><Image className="rounded-full" width={80} height={80} src={item.avatar_url + ".webp&s=80"} alt={item.login}></Image></a>
```

File: Website/app/Contribute.js [hunk 5]
```diff
@@ -27,18 +27,24 @@ export default function Contribute( {props} ) {
                 </p>
             </div>
             <div className="md:w-1/2">
-            <h2 className="text-5xl sm:text-4xl font-bold md:mt-20 md:mb-20 md:pl-20 xs:text-center xs:mb-10" >Be a part of Drifty Family!</h2>
-                <div className="grid lg:grid-cols-7 md:grid-cols-5 sm:grid-cols-8 xs:grid-cols-4 space-x-0 gap-y-4 justify-items-center xs:pb-10">
-                    {props.contrib.map((item,index) => {
-                        return <a  href={item.html_url} key={index}><Image width={64} height={64} src={item.avatar_url + ".webp&s=64"} alt={item.login}></Image></a>
+                <h2 className="text-5xl sm:text-4xl my-12 font-bold lg:mt-20 md:mt-10 sm:pt-10 sm:mb-10 md:pl-20 xs:p-5 xs:text-center" >Be a part of Drifty Family!</h2>
```
Comment (SaptarshiSarkar12) [inline Website/app/Contribute.js:30]: @PriyansuMaurya Please reduce the margin from `20` to `16`.
```suggestion
                <h2 className="text-5xl sm:text-4xl my-12 font-bold lg:mt-16 md:mt-10 sm:pt-10 sm:mb-10 md:pl-20 xs:p-5 xs:text-center" >Be a part of Drifty Family!</h2>
```

**Review by SaptarshiSarkar12 — CHANGES_REQUESTED** [review]
@PriyansuMaurya Along with the below changes, please change the `lg:mt-20` attribute in line number 9 to `lg:mt-16`.

File: Website/app/Contribute.js [hunk 6]
```diff
@@ -27,18 +27,24 @@ export default function Contribute( {props} ) {
...
-                    <a className="md:rounded-full xs:p-2 lg:p-2 lg:w-52 border-2 md:border-white text-center   hover:text-black xs:transition ease-in-out delay-150 bg-blue-500 hover:-translate-y-1 hover:scale-110 hover:bg-indigo-500 duration-300" href="https://github.com/SaptarshiSarkar12/Drifty/network/members"><i aria-hidden className="fa fa-duotone fa-code-fork pr-1"></i>Forks : {values[1]}</a>
+                <div class="flex flex-cols-3 justify-around items-center xs:gap-7 text-lg font-semibold mx-3">
+                    <a class="rounded-lg md:rounded-full xs:p-2 lg:p-2 lg:w-52 border-2 md:border-white text-center hover:text-black xs:transition ease-in-out delay-150 bg-blue-500 hover:-translate-y-1 hover:scale-110 hover:bg-indigo-500 duration-300" href="https://github.com/SaptarshiSarkar12/Drifty"><i aria-hidden="true" class="fab fa-github pr-1"></i>Contribute</a>
```
Comment (SaptarshiSarkar12) [inline Website/app/Contribute.js:42]:
```suggestion
                    <a class="rounded-lg xs:p-2 lg:p-2 lg:w-52 border-2 md:border-white text-center hover:text-black xs:transition ease-in-out delay-150 bg-blue-500 hover:-translate-y-1 hover:scale-110 hover:bg-indigo-500 duration-300" href="https://github.com/SaptarshiSarkar12/Drifty"><i aria-hidden="true" class="fab fa-github pr-1"></i>Contribute</a>
```

File: Website/app/Contribute.js [hunk 7]
```diff
@@ -27,18 +27,24 @@ export default function Contribute( {props} ) {
...
+                    <a class="rounded-lg md:rounded-full xs:p-2 lg:p-2 lg:w-52 border-2 md:border-white text-center hover:text-black xs:transition ease-in-out delay-150 bg-blue-500 hover:-translate-y-1 hover:scale-110 hover:bg-indigo-500 duration-300" href="https://github.com/SaptarshiSarkar12/Drifty/stargazers"><i aria-hidden="true" class="fa fa-regular fa-star pr-1"></i>Stars : 80</a>
```
Comment (SaptarshiSarkar12) [inline Website/app/Contribute.js:43]:
```suggestion
                    <a class="rounded-lg xs:p-2 lg:p-2 lg:w-52 border-2 md:border-white text-center hover:text-black xs:transition ease-in-out delay-150 bg-blue-500 hover:-translate-y-1 hover:scale-110 hover:bg-indigo-500 duration-300" href="https://github.com/SaptarshiSarkar12/Drifty/stargazers"><i aria-hidden="true" class="fa fa-regular fa-star pr-1"></i>Stars : 80</a>
```

File: Website/app/Contribute.js [hunk 8]
```diff
@@ -27,18 +27,24 @@ export default function Contribute( {props} ) {
...
+                    <a class="rounded-lg md:rounded-full xs:p-2 lg:p-2 lg:w-52 border-2 md:border-white text-center hover:text-black xs:transition ease-in-out delay-150 bg-blue-500 hover:-translate-y-1 hover:scale-110 hover:bg-indigo-500 duration-300" href="https://github.com/SaptarshiSarkar12/Drifty/network/members"><i aria-hidden="true" class="fa fa-duotone fa-code-fork pr-1"></i>Forks : 77</a>
```
Comment (SaptarshiSarkar12) [inline Website/app/Contribute.js:44]:
```suggestion
                    <a class="rounded-lg xs:p-2 lg:p-2 lg:w-52 border-2 md:border-white text-center hover:text-black xs:transition ease-in-out delay-150 bg-blue-500 hover:-translate-y-1 hover:scale-110 hover:bg-indigo-500 duration-300" href="https://github.com/SaptarshiSarkar12/Drifty/network/members"><i aria-hidden="true" class="fa fa-duotone fa-code-fork pr-1"></i>Forks : 77</a>
```

**Review by SaptarshiSarkar12 — CHANGES_REQUESTED** [review]
@PriyansuMaurya Please keep the partially rounded buttons by removing the `md:rounded-full` attribute.

**Review by SaptarshiSarkar12 — CHANGES_REQUESTED** [review]
@PriyansuMaurya In **IPad Mini**, the GitHub avatars are about to overlap. Can you please add some space/gap between two consecutive avatars (might be `gap-1`) ?
image

**Comment by PriyansuMaurya** [comment]
After a gap of two attempts, it was still not resolved. Therefore, I had to make changes to achieve what you wanted: 

`width={80} height={80}` => `width={64} height={64}`

If you believe that further changes are necessary, please note that I am not available for them. You can proceed to close this pull request.

**Comment by SaptarshiSarkar12** [comment]
> After a `gap-2`, it was still not resolved. Therefore, I had to make changes to achieve what you wanted:
> 
> `width={80} height={80}` => `width={64} height={64}`
> 
> If you believe that further changes are necessary, please note that I am not available for them. You can proceed to close this pull request.

@PriyansuMaurya It is fine if you have changed the image size. Also, no more changes are required for the IPad Mini case. But, there is some small syntactical and some logical changes required.

File: Website/app/Contribute.js [hunk 9]
```diff
@@ -27,18 +27,24 @@ export default function Contribute( {props} ) {
...
+                    <a class="rounded-lg xs:p-2 lg:p-2 lg:w-52 border-2 md:border-white text-center hover:text-black xs:transition ease-in-out delay-150 bg-blue-500 hover:-translate-y-1 hover:scale-110 hover:bg-indigo-500 duration-300" href="https://github.com/SaptarshiSarkar12/Drifty"><i aria-hidden="true" class="fab fa-github pr-1"></i>Contribute</a>
+                    <a class="rounded-lg xs:p-2 lg:p-2 lg:w-52 border-2 md:border-white text-center hover:text-black xs:transition ease-in-out delay-150 bg-blue-500 hover:-translate-y-1 hover:scale-110 hover:bg-indigo-500 duration-300" href="https://github.com/SaptarshiSarkar12/Drifty/stargazers"><i aria-hidden="true" class="fa fa-regular fa-star pr-1"></i>Stars : 80</a>
+                    <a class="rounded-lg xs:p-2 lg:p-2 lg:w-52 border-2 md:border-white text-center hover:text-black xs:transition ease-in-out delay-150 bg-blue-500 hover:-translate-y-1 hover:scale-110 hover:bg-indigo-500 duration-300" href="https://github.com/SaptarshiSarkar12/Drifty/network/members"><i aria-hidden="true" class="fa fa-duotone fa-code-fork pr-1"></i>Forks : 77</a>
```
Comment (SaptarshiSarkar12) [inline Website/app/Contribute.js:44]: @PriyansuMaurya Please change `class` to `className` and make the stars and forks dynamically determined as was present previously by committing this suggestion.
 ```suggestion
                <div className="flex flex-cols-3 justify-around items-center xs:gap-7 text-lg font-semibold mx-3">
                    <a className="rounded-lg xs:p-2 lg:p-2 lg:w-52 border-2 md:border-white text-center hover:text-black xs:transition ease-in-out delay-150 bg-blue-500 hover:-translate-y-1 hover:scale-110 hover:bg-indigo-500 duration-300" href="https://github.com/SaptarshiSarkar12/Drifty"><i aria-hidden="true" className="fab fa-github pr-1"></i>Contribute</a>
                    <a className="rounded-lg xs:p-2 lg:p-2 lg:w-52 border-2 md:border-white text-center hover:text-black xs:transition ease-in-out delay-150 bg-blue-500 hover:-translate-y-1 hover:scale-110 hover:bg-indigo-500 duration-300" href="https://github.com/SaptarshiSarkar12/Drifty/stargazers"><i aria-hidden="true" className="fa fa-regular fa-star pr-1"></i>Stars : {values[0]}</a>
                    <a class="rounded-lg xs:p-2 lg:p-2 lg:w-52 border-2 md:border-white text-center hover:text-black xs:transition ease-in-out delay-150 bg-blue-500 hover:-translate-y-1 hover:scale-110 hover:bg-indigo-500 duration-300" href="https://github.com/SaptarshiSarkar12/Drifty/network/members"><i aria-hidden="true" className="fa fa-duotone fa-code-fork pr-1"></i>Forks : {values[1]}</a>
```

**Review by SaptarshiSarkar12 — APPROVED** [review]

**Review by SaptarshiSarkar12 — APPROVED** [review]
@PriyansuMaurya **Looks good to merge 👍.** 
**Thanks for contributing 🚀 🚀.**
You may join our _Discord server_ - **https://discord.gg/DeT4jXPfkG** to get updates about the project.

**Pull request merged**